lib/travis/build/addons/mariadb.rb

```diff
@@ -29,6 +29,15 @@ def after_prepare
         def mariadb_version
           config.to_s.shellescape
         end
+
+        def mariadb_client
+          if config >= 10.2
```

Comment:
You are passing a String in specs, but comparing it with a Float. You should normalize it.

Also, if you are considering these inputs to represent software versions, it is better to treat it as such; e.g., with `Gem::Version`.